x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetJobsStatsAction.java

```diff
@@ -185,6 +190,7 @@ public JobStats(StreamInput in) throws IOException {
                 assignmentExplanation = in.readOptionalString();
                 openTime = in.readOptionalTimeValue();
                 forecastStats = in.readOptionalWriteable(ForecastStats::new);
+                timingStats = in.readOptionalWriteable(TimingStats::new);
```

Comment:
This needs to be version-aware so it works in a mixed version cluster:

```
    if (in.getVersion().onOrAfter(V_7_3_0)) {
        timingStats = in.readOptionalWriteable(TimingStats::new);
    } else {
        timingStats = null;
    }
```

Reply:
@droberts195 is right. Additionally, making it `onOrAfter(V_7_3_0)` immediately could break the bwc tests as the actual v7.3.0 code has not been backported to yet.

Reply:
Additionally, since this is the first BWC issue JobStats serialization, BWC should be added for job stats.

Reply:
@przemekwitek you should be able to add them to the BWC YAML tests. These are in `xpack.qa.rolling-upgrade.resources`. Probably cases added in `30_ml_jobs_crud.yml`

Reply:
I discussed BWC tests with @dimitris-athanasiou and he suggested that it will be easier if I add them (in a separate PR) after merging in this PR into master and backporting into 7.x. WDYT?

Reply:
Yes, it will be much easier to add the BWC tests in a follow-up PR.  Otherwise you'd have to add them on the basis that 7.x doesn't contain this functionality, then adjust both master and 7.x once 7.x does contain this functionality.  So I agree with the idea of adding them in a separate PR.